### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by imqt**
1. Added routing to registration page from App.tsx and LoginForm index.tsx
2. Added Registration page following how Login was done.
3. Added a bit of css for the registration form.

### CODE REVIEW

**Comment by BuzzBumble** [comment]
Created a Pull Request to Registration branch with small refactor:
https://github.com/webhook-buddies/webhook-buddy-client/pull/2

File: src/App.tsx [hunk 1]
```diff
@@ -16,7 +23,16 @@ function App() {
   const { data } = useQuery<IsUserLoggedIn>(IS_LOGGED_IN);
   const me = useMe();
   return !data || !data.isLoggedIn ? (
-    <Login />
+    <Router>
+      <Switch>
+        <Route exact path="/">
```
Comment (johnnyoshika) [inline src/App.tsx]: We should fall back to Login route by default if the `/register` path is not met. That way, if someone is on a specific URL (e.g. http://localhost:3000/#/foo/bar), we will still show the login form.

**Review by johnnyoshika — CHANGES_REQUESTED** [review]

**Comment by johnnyoshika** [comment]
@imqt I'm not able to load the project. Here's the error I get:

image

File: src/components/Register/RegisterForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  const [firstName, setFirstName] = useState<string>('');
+  const [lastName, setLastName] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [confirmPassword, setConfirmPassword] = useState<string>('');
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (password === confirmPassword) {
+      console.log('Send data to server!');
```
Comment (johnnyoshika) [inline src/components/Register/RegisterForm/index.tsx]: Remove console.log

File: src/components/Register/RegisterForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+          input: {
+            firstName,
+            lastName,
+            email,
+            password,
+          },
+        },
+      });
+    } else {
+      console.log('Passwords do not match!');
```
Comment (johnnyoshika) [inline src/components/Register/RegisterForm/index.tsx]: Remove console.log

File: src/components/Register/RegisterForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            </div>
+            <div className={`form-group ${styles.formGroupCustom}`}>
+              <label
+                className={
+                  confirmPassword
+                    ? `${styles.floatLabel}`
+                    : `${styles.noLabel}`
+                }
+              >
+                Confirm Password:
```
Comment (johnnyoshika) [inline src/components/Register/RegisterForm/index.tsx]: Let's remove confirm password. It'll be less things for the user to enter and confirming password is becoming more and more rare.

File: src/routes/Viewport/AutoForwarder/index.tsx [hunk 5]
```diff
@@ -70,7 +70,9 @@ const AutoForwarder = ({ docked }: { docked: Boolean }) => {
           endpointId={endpointId}
         />
       ) : (
-        <p>No endpoint selected.</p>
+        <div className={`${styles['not-selected']}`}>
+          <p>No Endpoint selected</p>
+        </div>
```
Comment (johnnyoshika) [inline src/routes/Viewport/AutoForwarder/index.tsx]: Why is this here? This seems unrelated to this PR.

File: src/components/Register/RegisterForm/styles.module.css [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  transition-duration: 0.3s;
+}
+
+.noLabel {
+  opacity: 0;
+  position: relative;
+  top: 40px;
+  transform: translateY(0px);
+  transition-duration: 0.3s;
+}
```
Comment (johnnyoshika) [inline src/components/Register/RegisterForm/styles.module.css:46]: A very nice touch to have the labels jump above as the user starts to type. 👍

**Review by johnnyoshika — CHANGES_REQUESTED** [review]
Looking really good @imqt! Just a few more tweaks and it'll be ready to merge.

File: src/components/Register/RegisterForm/styles.module.css (hunk 6 above)
Comment (imqt) [reply src/components/Register/RegisterForm/styles.module.css:46]: Thank you, i felt pretty proud about that one

File: src/routes/Viewport/AutoForwarder/index.tsx (hunk 5 above)
Comment (imqt) [reply src/routes/Viewport/AutoForwarder/index.tsx]: i forgot to take it out, sorry about that

File: src/components/Register/RegisterForm/index.tsx (hunk 4 above)
Comment (imqt) [reply src/components/Register/RegisterForm/index.tsx]: okay i'll remove that and push now

File: src/routes/Viewport/AutoForwarder/index.tsx (hunk 5 above)
Comment (johnnyoshika) [reply src/routes/Viewport/AutoForwarder/index.tsx]: Will you be taking it out? This seems like the last issue before this PR gets merged.
